contrib/win32/win32compat/shell-host.c

```diff
@@ -1280,10 +1308,24 @@ start_with_pty(wchar_t *command)
 	GOTO_CLEANUP_ON_FALSE(SetHandleInformation(pipe_in, HANDLE_FLAG_INHERIT, 0));
 	
 	cmd[0] = L'\0';
-	GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, w32_cmd_path()));	
+	GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, get_default_shell_path()));
 	if (command) {
-		GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, L" /c "));		
+		if(is_bash_default_shell)
+			GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, L" -c "));
+		else
+			GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, L" /c "));
+
 		GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, command));
+	} else {
+		/* If the user configures the default shell then launch it through cmd.exe */
+		if (is_default_shell_configured) {
+			wchar_t tmp_cmd[PATH_MAX + 1] = {0,};
+			wcscat_s(tmp_cmd, _countof(tmp_cmd), cmd);
+			cmd[0] = L'\0';
+			GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, w32_cmd_exe_path));
+			GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, L" /c "));
+			GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, tmp_cmd));
```

Comment:
If we don't launch the default shell from the cmd.exe then the console colors are very weird in case of powershell so we want to launch them through cmd.exe